File: rb/features.rb
# Kanyerest SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module KanyerestFeatures
  def self.make_feature(name)
    case name
    when "base"
      KanyerestBaseFeature.new
    when "ratelimit"
      KanyerestRatelimitFeature.new
    when "retry"
      KanyerestRetryFeature.new
    when "test"
      KanyerestTestFeature.new
    when "timeout"
      KanyerestTimeoutFeature.new
    else
      KanyerestBaseFeature.new
    end
  end
end

File: py/features.py
# Kanyerest SDK feature factory

from feature.base_feature import KanyerestBaseFeature
from feature.test_feature import KanyerestTestFeature


def _make_feature(name):
    features = {
        "base": lambda: KanyerestBaseFeature(),
        "test": lambda: KanyerestTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
